edelphi/src/main/typescript/src/components/panel/query-comment-list.tsx

```diff
@@ -1,86 +1,432 @@
+      pageId: pageId,
+      parentId: 0,
+      categoryId: categoryId,
+      firstResult: (activePage - 1) * commentsPerPage,
+      maxResults: commentsPerPage,
+      oldestFirst: oldestFirst
+    });
+
+    const totalCount = parseInt(response.raw.headers.get("X-Total-Count") || "0") || 0;
+    const comments = await response.value();
+
+    this.setState({
+      pageCount: Math.ceil(totalCount / commentsPerPage),
+      comments: comments.sort(CommentUtils.compareComments),
+      loading: false
     });
   }
-  
+
+  /**
+   * Loads child comments with parent ID
+   *
+   * @param parentId parent ID
+   */
+  private loadComments = async (parentId: number) => {
+    const { pageId, panelId, queryId, accessToken, category } = this.props;
+    const { comments } = this.state;
+
+    if (!accessToken) {
+      return;
+    }
+
+    const categoryId = category ? category.id : 0;
+    try {
+      const childComments = await Api.getQueryQuestionCommentsApi(accessToken).listQueryQuestionComments({
+        panelId: panelId,
+        queryId: queryId,
+        pageId: pageId,
+        parentId: parentId,
+        categoryId: categoryId,
+        firstResult: 0,
+        maxResults: 1000,
+        oldestFirst: false
+      });
+
+      const combinedComments = [ ...comments ];
+      childComments.forEach(
+        childComment => {
+          const childIndex = combinedComments.findIndex(comment => comment.id === childComment.id);
+          if (childIndex > -1) {
+            combinedComments.splice(childIndex, 1, childComment);
+          } else {
+            combinedComments.push(childComment);
+          }
+        }
+      )
+
+      this.setState({
+        comments: combinedComments
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  /**
+   * Handles query question comment notification MQTT message
+   *
+   * @param message message
+   */
+  private onQueryQuestionCommentNotification = async (message: QueryQuestionCommentNotification) => {
+    const { pageId, panelId } = this.props;
+
+    if (message.pageId !== pageId || message.panelId !== panelId) {
+      return;
+    }
+
+    const { commentId, parentCommentId, type } = message;
+
+    switch (type) {
+      case "CREATED":
+        this.onCreate(commentId);
+        break;
+      case "UPDATED":
+        this.onUpdate(commentId);
+        break;
+      case "DELETED":
+        this.onDelete(commentId, parentCommentId);
+        break;
+    }
+  }
+
+  /**
+   * Event handler for CREATED MQTT message
+   *
+   * @param commentId comment ID
+   */
+  private onCreate = async (commentId: number) => {
+    const { panelId, accessToken } = this.props;
+    const { comments } = this.state;
+
+    const commentList = [ ...comments ];
+
+    try {
+      const newComment = await Api.getQueryQuestionCommentsApi(accessToken).findQueryQuestionComment({
+        panelId: panelId,
+        commentId: commentId
+      });
+
+      if (newComment.parentId) {
```

Comment:
Shouldn't selected order affect placement of the new comment (first or last) on the comment list?